### PR TITLE
Final bringup

### PR DESCRIPTION
**Pull request by jr1221**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: Core/Src/control.c [hunk 1]
```diff
@@ -40,15 +42,20 @@ static void control_device(device_control_t *device, uint16_t temp)
...
-	if (device->device_type == DEVICE_PUMP0 ||
-	    device->device_type == DEVICE_PUMP1) {
-		if (hv) {
-			set_device_on(device);
-			return;
-		} else if (get_func_state() == FAULTED) {
-			set_device_off(device);
-			return;
-		}
+	// if (device->device_type == DEVICE_PUMP0 ||
```
Comment (Peyton-McKee) [inline Core/Src/control.c:46]: are we leaving this commented to save battery?

File: Core/Src/control.c [hunk 2]
```diff
@@ -40,15 +42,20 @@ static void control_device(device_control_t *device, uint16_t temp)
...
+	// 		set_device_on(device);
+	// 		return;
+	// 	} else
+	if (get_func_state() == FAULTED) {
+		set_device_off(device);
+		return;
+	}
+
+	uint16_t upper_temp = device->upper_temp;
+	if (!hv) {
```
Comment (Peyton-McKee) [inline Core/Src/control.c]: what is this code,

File: Core/Src/dti.c [hunk 3]
```diff
@@ -61,15 +73,13 @@ void dti_set_torque(int16_t torque)
 	}
 	float average = sum / SAMPLES;
 
-	if (torque == 0) {
-		average = 0;
-	}
+	// if (torque == 0) {
```
Comment (Peyton-McKee) [inline Core/Src/dti.c]: are we leaving this commented, i did this to prevent some pedal instability from spiking torque, it seems to be fine with the frequency of our pedals and torque setting

File: Core/Src/pdu.c [hunk 4]
```diff
@@ -68,6 +75,87 @@ int init_ina(pdu_t *pdu, ina226_t *ina, uint16_t dev_addr, float r_shunt,
...
 }
 
+bool verify_tca_config(pdu_t *pdu)
+{
+	assert(pdu);
+	assert(pdu->shutdown_expander);
+	assert(pdu->ctrl_expander);
+
+	uint8_t buf;
+	HAL_StatusTypeDef status = tca9539_read_reg(
```
Comment (Peyton-McKee) [inline Core/Src/pdu.c:85]: we should also read output registers, we set those as well I believe.

File: Core/Src/pedals.c [hunk 5]
```diff
@@ -126,28 +148,39 @@ void calc_pedal_faults(float accel1, float accel2, uint16_t accel1_norm,
...
+	bool short_circuit = accel1 < MIN_APPS1_VOLTS - APPS_THRESHOLD_BUF ||
+			     accel2 < MIN_APPS2_VOLTS - APPS_THRESHOLD_BUF;
 	debounce(short_circuit, &sc_fault_timer, PEDAL_FAULT_TIME,
 		 &pedal_fault_cb,
-		 "Pedal short circuit fault - no acceleration value");
+		 "Pedal grounded circuit fault - no acceleration value");
 
 	/* Pedal difference fault evaluation */
-	// Fault registered when more than 10% is detected between the sensor readings
+	// Fault registered when greater than 10% is detected between the sensor readings
```
Comment (Peyton-McKee) [inline Core/Src/pedals.c:165]: really feel we should up the percentage to 20. rules limit is 30 right

File: Core/Src/pedals.c [hunk 6]
```diff
@@ -428,44 +473,62 @@ void vProcessPedals(void *pv_params)
...
 		/* Turn brakelight on or off */
-		write_brakelight(pdu, brake_value > PEDAL_BRAKE_THRESH);
+
+		osMutexAcquire(brake_state_mut, osWaitForever);
+		if (brake_value > PEDAL_BRAKE_THRESH) {
+			brake_pressed = true;
+		} else {
+			brake_pressed = false;
+		}
+		osMutexRelease(brake_state_mut);
```
Comment (Peyton-McKee) [inline Core/Src/pedals.c:514]: we should release mutex after writing brakelight i think

File: Core/Src/control.c (hunk 2 above)
Comment (jr1221) [reply Core/Src/control.c]: worth a comment.  its basically the pumps should be on when the car is not driving more often then when it is.

File: Core/Src/dti.c (hunk 3 above)
Comment (jr1221) [reply Core/Src/dti.c]: might want it

File: Core/Src/pedals.c (hunk 5 above)
Comment (jr1221) [reply Core/Src/pedals.c:165]: believe it is 20 lol

File: Core/Src/control.c (hunk 2 above)
Comment (Peyton-McKee) [reply Core/Src/control.c]: why would we want it on more when its not driving?

File: Core/Src/pdu.c (hunk 4 above)
Comment (Peyton-McKee) [reply Core/Src/pdu.c:85]: diregard this

File: Core/Src/pedals.c [hunk 7]
```diff
@@ -428,44 +473,62 @@ void vProcessPedals(void *pv_params)
...
-		uint16_t accel2_norm = pedal_percent_pressed(
-			accel2_volts, APPS2_VOLTAGE_OFFSET, MAX_APPS2_VOLTS);
+		float accel1_norm = pedal_percent_pressed(
+			accel1_volts, MIN_APPS1_VOLTS, MAX_APPS1_VOLTS);
+		float accel2_norm = pedal_percent_pressed(
+			accel2_volts, MIN_APPS2_VOLTS, MAX_APPS2_VOLTS);
 
-		calc_pedal_faults(accel1_volts, accel2_volts, accel1_norm,
-				  accel2_norm);
+		// bool possible_faults = calc_pedal_faults(
```
Comment (Peyton-McKee) [inline Core/Src/pedals.c]: probably should uncomment lol
